File: codegen/results/v002/lmstudio/qwen3.8-27b-mlx_think/20260828_101748/python/004_json_filter/solution.py
import json
import sys

with open("input/data.json", "r", encoding="utf-8") as f:
    records = json.load(f)

selected = [
    record
    for record in records
    if record.get("active") is True and record.get("age", 0) >= 30
]

selected.sort(key=lambda record: record.get("name", ""))
sys.stdout.write(json.dumps(selected))

File: codegen/results/v002/lmstudio/qwen3.8-27b-mlx_think/20260828_101748/typescript/004_json_filter/solution.ts
import * as fs from "node:fs";

interface Person {
  name: string;
  age: number;
  active: boolean;
  score: number;
}

const filePath = process.argv[2] ?? "input/data.json";
const data = JSON.parse(fs.readFileSync(filePath, "utf8")) as Person[];

const result = data
  .filter((person) => person.active === true && person.age >= 30)
  .sort((a, b) => {
    if (a.name < b.name) return -1;
    if (a.name > b.name) return 1;
    return 0;
  });

process.stdout.write(JSON.stringify(result));